src/components/AddItemForm.js

```diff
@@ -1,28 +1,60 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import fb from '../lib/firebase';
 import '../css/AddItemForm.css';
-import { v4 as uuidv4 } from 'uuid';
+// import { Redirect } from 'react-router-dom';
 
 const Form = ({token}) => {
     const [itemName, setItemName] = useState("");
     const [timeFrame, setTimeFrame] = useState(7);
     const [lastPurchaseDate, setPurchaseDate] = useState(null);
     const userToken = token || "userToken";
+    const [shoppingListCollection, setShoppingListCollection] = useState([]);
+    const [duplicateError, setDuplicateError] = useState(false);
+
+    useEffect(() => {
+        const db =  fb.firestore()
+        const tokenRef = db.collection(token);
+
+        tokenRef
+            .orderBy("timeFrame", "asc")
+            .get()
+            .then((querySnapshot) => {
+                let fullCollection = [];
+
+                querySnapshot.forEach((doc) => {
+                    let documentData = doc.data();
+                    let nameData = documentData.itemName
+                    nameData = nameData.toLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g,"").trim().replace(/\s{2,}/g," ");
+
```

Comment:
Did this come from somewhere?  How does this work if there is nothing in the db? Like if there is no nameData, it might error on `nameData.toLowerCase()`, I wonder.

Reply:
After you successfully add an item, does this useEffect run again? We would want it to, so that your `shoppingListCollection` is updated.

Reply:
> Did this come from somewhere? How does this work if there is nothing in the db? Like if there is no nameData, it might error on `nameData.toLowerCase()`, I wonder.

That's a good point. We should put in a conditional to make sure the db isn't empty.